source/JABHandler.py

```diff
@@ -701,37 +701,40 @@ def internal_hasFocus(sourceContext):
 @AccessBridge_PropertyNameChangeFP
 def event_nameChange(vmID,event,source,oldVal,newVal):
 	jabContext=JABContext(vmID=vmID,accContext=source)
-	focus=api.getFocusObject()
-	if isinstance(focus, NVDAObjects.JAB.JAB) and focus.jabContext == jabContext:
-		obj = focus
-	else:
-		obj = NVDAObjects.JAB.JAB(jabContext=jabContext)
-	if obj:
-		eventHandler.queueEvent("nameChange", obj)
+	if jabContext.hwnd:
+		focus = api.getFocusObject()
+		if isinstance(focus, NVDAObjects.JAB.JAB) and focus.jabContext == jabContext:
+			obj = focus
+		else:
+			obj = NVDAObjects.JAB.JAB(jabContext=jabContext)
+		if obj:
+			eventHandler.queueEvent("nameChange", obj)

```

Comment:
Instead of increasing the indentation depth, I suggest returning early and logging a warning for each of these

```suggestion
	if jabContext.hwnd:
		log.debugWarning("Unable to find JAB Window")
		return
	focus = api.getFocusObject()
	if isinstance(focus, NVDAObjects.JAB.JAB) and focus.jabContext == jabContext:
		obj = focus
	else:
		obj = NVDAObjects.JAB.JAB(jabContext=jabContext)
	if obj:
		eventHandler.queueEvent("nameChange", obj)
```